service/Microsoft.Azure.Devices/RegistryManager.cs

```diff
@@ -634,5 +634,33 @@ public void Dispose()
         /// <param name="cancellationToken">Cancellation token</param>
         /// <returns>Updated Twin instance</returns>
         public abstract Task<Twin> ReplaceTwinAsync(string deviceId, string newTwinJson, string etag, CancellationToken cancellationToken);
+
+        /// <summary>
+        /// Update the diagnostic settings
+        /// </summary>
+        /// <param name="deviceId">Device Id</param>
+        /// <param name="setting"><see cref="E2EDiagnosticSetting"/></param>
+        /// <param name="etag">Device ETag</param>
+        /// <returns>echoes back the E2EDiagnosticSetting object</returns>
```

Comment:
echoes back the E2EDiagnosticSetting object
-->Echo

Reply:
fixed